### ISSUE TITLE
Consider move Device Twin's Desired field to spec

### ISSUE THREAD

**Issue by fisherxu** [kubeedge/kubeedge#679]
**What would you like to be updated**:

Now the Device Twin's Desired field [L134](https://github.com/kubeedge/kubeedge/blob/master/cloud/pkg/devicecontroller/apis/devices/v1alpha1/device_instance_types.go#L134) is in DeviceStatus, users operating on device must set this status field. And we also need to validate the status fields in CRD. 
 
In the design concept of kubernetes user should not update the status, so can we consider moving this to spec? :)

**Comment by sids-b** [comment]
cc @kevin-wangzefeng @rohitsardesai83 @chendave

**Comment by rohitsardesai83** [comment]
yes , it sounds logical to move things which users want to control on device via spec and not status.
let's solicit more feedback from users / community ?

**Comment by sids-b** [comment]
@kevin-wangzefeng @edisonxiang @subpathdev  @chendave your thoughts ??

**Comment by subpathdev** [comment]
As described [here](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#object-spec-and-status), we should move the desired field to specs.
